go/test/endtoend/vtgate/schema/schema_test.go

```diff
@@ -0,0 +1,290 @@
+func matchSchema(t *testing.T, firstTablet string, secondTablet string) {
+	firstShardSchema, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("GetSchema", firstTablet)
+	assert.Nil(t, err)
+
+	secondShardSchema, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("GetSchema", secondTablet)
+	assert.Nil(t, err)
+
+	assert.Equal(t, firstShardSchema, secondShardSchema)
+}
+
+// testSchemaChangePreflightErrorPartialy applying same schema + new schema should throw error for existing one
+// Tests that some SQL statements fail properly during PreflightSchema.
+func testSchemaChangePreflightErrorPartialy(t *testing.T) {
+	createNewTable := fmt.Sprintf(createTable, fmt.Sprintf("vt_select_test_%02d", 5)) + fmt.Sprintf(createTable, fmt.Sprintf("vt_select_test_%02d", 2))
+	output, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("ApplySchema", "-sql", createNewTable, keyspaceName)
+	assert.NotNil(t, err)
+	assert.True(t, strings.Contains(output, "already exists"))
+
+	checkTables(t, totalTableCount)
+}
+
+// testSchemaChangePreflightErrorPartialy applying same schema + new schema should throw error for existing one and also add the new schema
+//Tests the PreflightSchema logic for dropping nonexistent tables.
+//
+//If a table does not exist, DROP TABLE should error during preflight
+//because the statement does not change the schema as there is
+//nothing to drop.
+//In case of DROP TABLE IF EXISTS though, it should not error as this
+//is the MySQL behavior the user expects.
+func testDropNonExistentTables(t *testing.T) {
+	dropNonExistentTable := "DROP TABLE nonexistent_table;"
+	output, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("ApplySchema", "-sql", dropNonExistentTable, keyspaceName)
+	assert.NotNil(t, err)
+	assert.True(t, strings.Contains(output, "Unknown table"))
+
+	dropIfExists := "DROP TABLE IF EXISTS nonexistent_table;"
+	err = clusterInstance.VtctlclientProcess.ApplySchema(keyspaceName, dropIfExists)
+	assert.Nil(t, err)
+
+	checkTables(t, totalTableCount)
+}
+
+// checkTables check if first 2 tablets' table count match
+func checkTables(t *testing.T, count int) {
+	checkTablesForCount(t, clusterInstance.Keyspaces[0].Shards[0].Vttablets[0], count)
+	checkTablesForCount(t, clusterInstance.Keyspaces[0].Shards[1].Vttablets[0], count)
+}
+
+// checkTablesForCount makes sql query to match supplied tablet's table
+func checkTablesForCount(t *testing.T, tablet cluster.Vttablet, count int) {
+	queryResult, err := tablet.VttabletProcess.QueryTablet("show tables;", keyspaceName, true)
+	assert.Nil(t, err)
+	assert.Equal(t, len(queryResult.Rows), count)
+}
+
+// testCopySchemaShards copy schema from source should be applied to destination and tables, schema should match
+func testCopySchemaShards(t *testing.T, source string, shard int) {
+	addNewShard(t, shard)
+	// InitShardMaster creates the db, but there shouldn't be any tables yet.
```

Comment:
I don't think so. Maybe you meant InitTablet?

Reply:
I have taken all these comments from existing schema.py, just so that we don't miss those comments, anyway, initshardmaster created db as per [this](https://planetscale.slack.com/archives/DPH8GA4F2/p1571849223010600).

Reply:
Haha. OK this stays as is.